### PR TITLE
DOC: Added an API contract for /extension-requests/user/:userId

### PR DESCRIPTION
**Pull request by vikasosmium**
Date: 18-DEC-2024

Developer Name: @vikasosmium 

---

## Issue Ticket Number

- Reference PR [#2300](https://github.com/Real-Dev-Squad/website-backend/pull/2300)

## Description

I have added an API contract for a new route /extension-requests/user/:userId.

### Documentation Updated?

- [x] Yes
- [ ] No

### Under Feature Flag

- [x] Yes
- [ ] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

</details>

### CODE REVIEW

File: extension-requests/README.md [hunk 1]
```diff
@@ -128,6 +129,39 @@ Returns all the extension requests of a user for a task if query `taskId=<taskId
     - **Content:** `{ 'statusCode': 500, 'error': 'Internal Server Error', 'message': 'An internal server error occurred' }`
 
 
+## **GET /extension-requests/user/:userId**
+
+Returns all the extension requests of a authenticated user for a task by userId if query `taskId=<taskId>` is passed, else returns all the extension requests of the user.
+
+- **Params**  
+    userId=<userId>
```
Comment (Mir-SA) [inline extension-requests/README.md:137]: can you add `` around < userId >? it highlights the word, it can be convenient for reading...

Comment (vikasosmium) [reply extension-requests/README.md:137]: done

File: extension-requests/README.md [hunk 2]
```diff
@@ -128,6 +129,39 @@ Returns all the extension requests of a user for a task if query `taskId=<taskId
...
+- **Query**  
+  `status=APPROVED | DENIED | PENDING, taskId=<task_object_id>`
```
[the remark this answers is not in the document]
Comment (vikasosmium) [reply extension-requests/README.md:139]: optional params

File: extension-requests/README.md [hunk 3]
```diff
@@ -128,6 +129,39 @@ Returns all the extension requests of a user for a task if query `taskId=<taskId
...
+}
+```
+
+- **Error Response:**
+  - **Code:** 401
+    - **Content:** `{ 'statusCode': 401, 'error': 'Unauthorized', 'message': 'Unauthenticated User' }`
+  - **Code:** 404
+    - **Content:** `{ 'statusCode': 404, 'error': 'Not Found', 'message': 'User doesn't exist' }`
+  - **Code:** 500
+    - **Content:** `{ 'statusCode': 500, 'error': 'Internal Server Error', 'message': 'An internal server error occurred' }`
```
[the remark this answers is not in the document]
Comment (vikasosmium) [reply extension-requests/README.md]: added, thanx for reviewing.

File: extension-requests/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mir-SA) [reply extension-requests/README.md:139]: got it!

**Review by Mir-SA — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]

**Pull request merged**